zkevm-circuits/src/sig_circuit.rs

```diff
@@ -268,12 +331,14 @@ impl<F: Field> SubCircuit<F> for SigCircuit<F> {
             .iter()
             .filter(|tx| !tx.tx_type.is_l1_msg())
             .count()
-            + block.precompile_events.get_ecrecover_events().len();
+            + block.precompile_events.get_ecrecover_events().len()
+            + block.precompile_events.get_p256_verify_events().len();
+
         // Reserve one ecdsa verification for padding tx such that the bad case in which some tx
         // calls MAX_NUM_SIG - 1 ecrecover precompile won't happen. If that case happens, the sig
         // circuit won't have more space for the padding tx's ECDSA verification. Then the
         // prover won't be able to produce any valid proof.
-        let max_num_verif = MAX_NUM_SIG - 1;
+        let max_num_verif = MAX_NUM_SIG_K1 + MAX_NUM_SIG_R1 - 1;
```

Comment:
this is wrong.  i explained the result is not (num_k1 + num_r1) / (limit_k1 + limi_r1), but max(num_k1/limit_k1, num_r1/limi_r1)

Reply:
btw i don't know why the "-1" is here. you can read the codes for why there is a "-1".

Reply:
will change to max_num_verif =  max(num_k1/limit_k1, num_r1/limi_r1) - 2  as last tx may include both ecrecover and p256verify at the same time.